tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -42,26 +43,288 @@ limitations under the License.
+    src_md_.reset(new memory::desc({src_dims}, MklDnnType<T>(),
+                                   memory::format::any));
+
+    filter_md_.reset(new memory::desc({filter_dims},
+                  MklDnnType<T>(), memory::format::any));
+
+    dst_md_.reset(new memory::desc({dst_dims}, MklDnnType<T>(),
+                                   memory::format::any));
+    if (!bias_dims.empty())
+        bias_md_.reset(new memory::desc({bias_dims}, MklDnnType<T>(),
+                                   memory::format::any));
+    // create a convolution
+    if (!bias_dims.empty()) {
+      fwd_desc_.reset(new convolution_forward::desc(prop_kind::forward,
+                      convolution_direct, *src_md_, *filter_md_,
+                      *bias_md_, *dst_md_, strides_, dilations,
+                      padding_l_, padding_r_, padding_kind::zero));
+    } else {
+      fwd_desc_.reset(new convolution_forward::desc(prop_kind::forward,
+                      convolution_direct, *src_md_, *filter_md_,
+                      *dst_md_, strides_, dilations,
+                      padding_l_, padding_r_, padding_kind::zero));
+    }
+
+    fwd_pd_.reset(new convolution_forward::primitive_desc(
+                  *fwd_desc_, cpu_engine_));
+
+    // store the expected memory format
+    src_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->src_primitive_desc().desc().data.format);
+
+    filter_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->weights_primitive_desc().desc().data.format);
+    dst_fmt_ = static_cast<mkldnn::memory::format>(
+                 fwd_pd_.get()->dst_primitive_desc().desc().data.format);
+
+    // create memory primitive based on dummy data
+    src_mem_.reset(new memory(fwd_pd_.get()->src_primitive_desc(), DummyData));
+    filter_mem_.reset(new memory(fwd_pd_.get()->weights_primitive_desc(),
+                      DummyData));
+    dst_mem_.reset(new memory(fwd_pd_.get()->dst_primitive_desc(), DummyData));
+
+    // create convolution primitive and add it to net
+    if (!bias_dims.empty()) {
+        bias_mem_.reset(new memory({{{bias_dims}, MklDnnType<T>(),
+                                   memory::format::x}, cpu_engine_}, DummyData));
+        conv_fwd_.reset(new convolution_forward(*fwd_pd_, *src_mem_,
+                        *filter_mem_, *bias_mem_, *dst_mem_));
+    } else {
+        conv_fwd_.reset(new convolution_forward(*fwd_pd_, *src_mem_,
+                        *filter_mem_, *dst_mem_));
+    }
+
+    fwd_primitives_.push_back(*conv_fwd_);
+    return;
+  }
+
+ public:
+  // Convolution forward execute with bias
+  void Execute(void* src, void* w, void* b, void* dst) {
+    src_mem_->set_data_handle(src);
+    filter_mem_->set_data_handle(w);
+    bias_mem_->set_data_handle(b);
+    dst_mem_->set_data_handle(dst);
+    fwd_stream_->submit(fwd_primitives_);
+
+    // after exec, set data handle back
+    src_mem_->set_data_handle(DummyData);
```

Comment:
Why is this hackery necessary?

Reply:
This is a kind of safe-guard: after an MKL op finishes its computation ("execute" here), 
untie related tensor buffers to MKL primitives which may be reused. 

We will revisit if the "hackery" is necessary in future PR's  (primitive reuse for other ops).

Thanks.

Reply:
Actually those buffers, inputs to "Execute" method, are tensor buffers. TF framework shall release them. 
set_data_handle() call is to detach them from MKL DNN primitives after its computation, to avoid the possibility of using them (outdated) for next computation. 

src_mem_->set_data_handle(nullptr) does not work because MKL DNN primitives requirement that the buffer not nullptr. I have experimented with this.   Thanks!